code/game/machinery/big_manipulator/big_manipulator.dm

```diff
@@ -0,0 +1,909 @@
+
+	/// How many interaction points of each kind can we have?
+	var/interaction_point_limit = MAX_INTERACTION_POINTS_TIER_1
+	/// List of pickup points.
+	var/list/pickup_points = list()
+	/// List of dropoff points.
+	var/list/dropoff_points = list()
+
+	/// Which tasking scenario we use for pickup points?
+	var/pickup_tasking = TASKING_ROUND_ROBIN
+	/// Which tasking scenario we use for dropoff points?
+	var/dropoff_tasking = TASKING_ROUND_ROBIN
+
+	/// List of all HUD icons resembling interaction points.
+	var/list/hud_points = list()
+
+	/// Pickup strategy for tasking.
+	var/datum/tasking_strategy/pickup_strategy
+	/// Dropoff strategy for tasking.
+	var/datum/tasking_strategy/dropoff_strategy
+
+/obj/machinery/big_manipulator/proc/update_hud()
+  LAZYCLEARLIST(hud_points)
+
+  var/image/main_hud = hud_list[BIG_MANIP_HUD]
+  if(!main_hud)
+    return
+
+  main_hud.loc = get_turf(src)
+  main_hud.appearance = mutable_appearance('icons/effects/interaction_points.dmi', null, ABOVE_NORMAL_TURF_LAYER, src, src.plane)
+
+  main_hud.overlays.Cut()
+  var/list/point_overlays = list()
+
+  for(var/i = 1 in 1 to length(pickup_points))
+    var/datum/interaction_point/point = pickup_points[i]
+    var/turf/target_turf = point.interaction_turf
+    if(target_turf)
+      var/mutable_appearance/point_appearance = mutable_appearance('icons/effects/interaction_points.dmi', "pickup_[i]", ABOVE_NORMAL_TURF_LAYER, src, src.plane)
+      var/turf/manip_turf = get_turf(src)
+      point_appearance.pixel_x = (target_turf.x - manip_turf.x) * 32
+      point_appearance.pixel_y = (target_turf.y - manip_turf.y) * 32
+      point_overlays += point_appearance
+
+  for(var/i = 1 in 1 to length(pickup_points))
+    var/datum/interaction_point/point = dropoff_points[i]
+    var/turf/target_turf = point.interaction_turf
+    if(target_turf)
+      var/mutable_appearance/point_appearance = mutable_appearance('icons/effects/interaction_points.dmi', "dropoff_[i]", ABOVE_NORMAL_TURF_LAYER, src, src.plane)
+      var/turf/manip_turf = get_turf(src)
+      point_appearance.pixel_x = (target_turf.x - manip_turf.x) * 32
+      point_appearance.pixel_y = (target_turf.y - manip_turf.y) * 32
+      point_overlays += point_appearance
+
+  main_hud.overlays += point_overlays
+  hud_points += main_hud
+  set_hud_image_active(BIG_MANIP_HUD)
+
+/// Attempts to find the closest open turf to the manipulator
+/obj/machinery/big_manipulator/proc/find_suitable_turf()
+	var/turf/center = get_turf(src)
+
+	var/list/directions = GLOB.alldirs
+	for(var/dir in directions)
+		var/turf/checked_turf = get_step(center, dir)
+		if(checked_turf && !isclosedturf(checked_turf))
+			return checked_turf
+
+	// didn't find any :boowomp:
+	return null
+
+/// Attempts to create a new interaction point and assign it to the correct list.
+/// Arguments: `new_turf` (turf), `new_filters` (list), `new_filters_status` (boolean),
+/// `new_interaction_mode` (use a define), `transfer_type` (use a define).
```

Comment:
consider using the /** formatted comments, I think they'd display this better (see tgui autodoc, style was real into em)